azure_data_cosmos_engine/query-tests/src/runner.rs

```diff
@@ -381,47 +381,38 @@ pub async fn run_baseline_test(
         );
     }
 
-    // Get the list of properties to validate from the first item
-    let properties = items
-        .first()
-        .map(|i| i.keys().cloned().collect::<Vec<_>>())
-        .expect("expected at least one item in the results");
-    let mut validation_errors = Vec::new();
-    for property in properties {
-        // Run the validator for each property
-        let validator = get_validator(&test_query.validators, &property);
-        tracing::debug!(property, validator, "validating property");
-        let expected_items: Vec<_> = results
-            .iter()
-            .map(|item| item.get(&property).cloned().unwrap_or_default())
-            .collect();
-        let actual_items: Vec<_> = items
-            .iter()
-            .map(|item| item.get(&property).cloned().unwrap_or_default())
-            .collect();
-        let property_errors =
-            validate_property(validator, &property, &expected_items, &actual_items);
-        tracing::debug!(
-            property,
-            validator,
-            error_count = property_errors.len(),
-            "validated property"
-        );
-        validation_errors.extend(property_errors);
-    }
+    // If the first expected item is an object, we do per-property validation.
+    let errors = if let Some(serde_json::Value::Object(_)) = results.first() {
+        validate_with_validators(&test_query.validators, &items, &results)?
+    } else {
+        // Otherwise, we just compare the values directly.
+        let mut errors = Vec::new();
+        for (i, (actual, expected)) in items.iter().zip(results.iter()).enumerate() {
```

Comment:
How does `zip` work in Rust for collections of unequal lengths?

Reply:
It ends when either one of the iterators end. But in this case, we've already checked that the lengths are the same up a few lines from here.